### PR TITLE
`nft.trades`: add tofu_nft on Polygon

### PR DESCRIPTION
**Pull request by sohwak**
Brief comments on the purpose of your changes:

**For Dune Engine V2**

Adding tofu.nft trade data to a polygon

tofu.nft is multichain and we have data from other chains, but not yet for polygon.
So I referenced @theachenyj 's code and added the same.

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by sohwak** [comment]
Ok. I think this PR is ready to merge. Please review this.

File: models/tofu/polygon/tofu_polygon_events.sql [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                id,
+                get_json_object(inventory, '$.seller')   as seller,
+                get_json_object(inventory, '$.buyer')    as buyer,
+                get_json_object(inventory, '$.kind')     as kind,
+                get_json_object(inventory, '$.price')    as price,
+                CASE WHEN get_json_object(inventory, '$.currency') = '0x0000000000000000000000000000000000000000'
+                  THEN '{{MATIC_ADDRESS}}'
+                  ELSE get_json_object(inventory, '$.currency')
+                END as currency,
+                (get_json_object(inventory, '$.currency') = '0x0000000000000000000000000000000000000000') as native_eth,
```
Comment (0xRobin) [inline models/tofu/polygon/tofu_polygon_events.sql:60]:
```suggestion
                (get_json_object(inventory, '$.currency') = '0x0000000000000000000000000000000000000000') as native_matic,
```

Let's change this for more clarity.

File: models/tofu/polygon/tofu_polygon_events.sql [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+     , case
+           when tfe.kind = '1' then 'Buy'
+           when tfe.kind = '2' then 'Sell'
+           else 'Auction'
+    end                                      as trade_category
+     , CAST(tfe.price AS DECIMAL(38,0))      as amount_raw
+     , tfe.price / power(10, pu.decimals)    as amount_original
+     , pu.price * tfe.price / power(10, pu.decimals) as amount_usd
+     , case
+           when tfe.native_eth THEN 'MATIC'
```
Comment (0xRobin) [inline models/tofu/polygon/tofu_polygon_events.sql:94]:
```suggestion
           when tfe.native_matic THEN 'MATIC'
```

File: models/tofu/polygon/tofu_polygon_events.sql [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+     , CAST(tfe.price * tff.fee_rate / power(10, pu.decimals) AS DOUBLE)            as platform_fee_amount
+     , CAST(pu.price * tfe.price * tff.fee_rate / power(10, pu.decimals) AS DOUBLE) as platform_fee_amount_usd
+     , CAST(100 * tff.fee_rate AS DOUBLE)                                           as platform_fee_percentage
+     , tfe.price * tff.royalty_rate                                                 as royalty_fee_amount_raw
+     , tfe.price * tff.royalty_rate / power(10, pu.decimals)                        as royalty_fee_amount
+     , pu.price * tfe.price * tff.royalty_rate / power(10, pu.decimals)             as royalty_fee_amount_usd
+     , CAST(100 * tff.royalty_rate AS DOUBLE)                                       as royalty_fee_percentage
+     , tff.royalty_address                                                          as royalty_fee_receive_address
+     , case
+           when tfe.native_eth THEN 'MATIC'
```
Comment (0xRobin) [inline models/tofu/polygon/tofu_polygon_events.sql:115]:
```suggestion
           when tfe.native_matic THEN 'MATIC'
```

**Review by 0xRobin — APPROVED** [review]
:heavy_check_mark: Looks good, thanks for including a seed test! 

I'll commit the trivial change from `native_eth` -> `native_matic` and mark it as 'ready to merge'.

**Review by jeff-dude — APPROVED** [review]
pushed a minor final cleanup

**Pull request merged**